k4e-agent-sti-upgrade.sh

```diff
@@ -0,0 +1,194 @@
+#!/bin/bash
+
+set -e
+
+# The script performs the following steps on rhel-8.5:
+# * Create new rpms for yggdrasil, k4e-device-worker and Prometheus node_exporter
+# * Publish rpms as yum repo to be consumed by image-builder
+# * Publish rhel4edge image commit for edge-device
+# * ./auto-upgrade.sh -a <server_adress> -i <image_name> -o helios06.lab.eng.tlv2.redhat.com:8888 -v
+
+IMAGE_NAME=
+HTTP_API=
+VERBOSE=
+
+usage()
+{
+cat << EOF
+usage: $0 options
+This script will create an image file and ISO to be served for download.
+OPTIONS:
+   -h      Show this message
+   -a      Image server address
+   -i      Original Image name(required)
+   -o      Operator's HTTP API (required)
+   -v      Verbose
+EOF
+}
+
+while getopts "h:a:i:o:v" option; do
+    case "${option}"
+    in
+        h)
+            usage
+            exit 0
+            ;;
+        a) IMAGE_SERVER_ADDRESS=${OPTARG};;
+        i) IMAGE_NAME=${OPTARG};;
+        o) HTTP_API=${OPTARG};;
+        v) VERBOSE=1;;
+    esac
+done
+
+dnf install -y bind-utils
+if [[ -z $IMAGE_SERVER_ADDRESS ]]; then
+    hostname_output=$(host $(hostname))
+    for word in $hostname_output
+    do
+      if [[ ($word =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$)  && ($word != "127.0.0.1")]]; then
+        echo "found hostname"
+        IMAGE_SERVER_ADDRESS=$word
+        break
+      fi
+    done
+    if [[ -z $IMAGE_SERVER_ADDRESS ]]; then
+      echo "ERROR: server address is required"
+      usage
+      exit 1
+    fi
+fi
+if [[ -z $IMAGE_NAME ]]; then
+    echo "ERROR: Image name is required"
+    usage
+    exit 1
+fi
+
+if [[ -z $HTTP_API ]]; then
+    echo "ERROR: Operator's HTTP-API url is required"
+    usage
+    exit 1
+fi
+
+if [[ ! -z $VERBOSE ]]; then
+    echo "Building ${IMAGE_NAME} image for Operator's HTTP-API ${HTTP_API}"
+    set -xv
+fi
+
+# export for use in templates
+export IMAGE_NAME
+export HTTP_API
+export RAND_SUFFIX=$((1 + $RANDOM % 10))
+PACKAGES_REPO=k4e-repo-$RAND_SUFFIX
+PACKAGE_REPO_URL=http://$IMAGE_SERVER_ADDRESS/$PACKAGES_REPO
+BLUEPRINT_NAME=$IMAGE_NAME
+IMAGE_UPGRADE_FOLDER=/var/www/html/$IMAGE_NAME-upgrade
+IMAGE_UPGRADE_URL=http://$IMAGE_SERVER_ADDRESS/$IMAGE_NAME-upgrade
+
+#---------------------------------
+# Cleanup before running
+#---------------------------------
+rm -rf ~/rpmbuild/*
+rm -rf /var/www/html/k4e-repo*
+rm -rf /home/builder/yggdrasil
+rm -rf /home/builder/k4e-device-worker
+rm -rf /home/builder/prometheus-node_exporter-rpm
+rm -rf /var/www/html/$IMAGE_UPGRADE_FOLDER
+rm -rf /var/www/html/$IMAGE_NAME-upgrade
+# need to remove all sources
+composer-cli sources delete agent
+
+#-----------------------------------
```

Comment:
some of this script is the same as the k4e-sti.sh, some logic can be extracted to a common file name to be reused (functions to create yggdrasil/device-worker/node-exporter rpms, function to create source resource, ...)

Reply:
I already created a ticket for refactoring r4e (reusing code, removing unused files etc...)